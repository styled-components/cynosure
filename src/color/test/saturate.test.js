// @flow
import saturate from '../saturate'

describe('saturate', () => {
  it('should saturate a color by 10%', () => {
    expect(saturate(0.1, '#444')).toMatchSnapshot()
  })

  it('should saturate a hex color by 20%', () => {
    expect(saturate(0.2, '#CCCD64')).toMatchSnapshot()
  })

  it('should saturate an 8-digit hex color by 20%', () => {
    expect(saturate(0.2, '#6564CDB3')).toMatchSnapshot()
  })

  it('should saturate a color with opacity by 20%', () => {
    expect(saturate(0.2, 'rgba(101,100,205,0.7)')).toMatchSnapshot()
  })

  it('should saturate a color but not go beyond 255', () => {
    expect(saturate(0.8, 'rgba(255,200,200,0.7)')).toMatchSnapshot()
  })

  it('should saturate a color when passed a string for amount', () => {
    expect(saturate('0.1', '#444')).toMatchSnapshot()
  })
})
