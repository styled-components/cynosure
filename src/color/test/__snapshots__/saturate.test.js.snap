// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`saturate should saturate a color but not go beyond 255 1`] = `"rgba(255,200,200,0.7)"`;

exports[`saturate should saturate a color by 10% 1`] = `"#4b3d3d"`;

exports[`saturate should saturate a color when passed a string for amount 1`] = `"#4b3d3d"`;

exports[`saturate should saturate a color with opacity by 20% 1`] = `"rgba(81,80,226,0.7)"`;

exports[`saturate should saturate a hex color by 20% 1`] = `"#e0e250"`;

exports[`saturate should saturate an 8-digit hex color by 20% 1`] = `"rgba(81,80,226,0.7)"`;
